client-id/store/s3_test.go

```diff
@@ -0,0 +1,72 @@
+package store
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"sync"
+	"testing"
+
+	"github.com/SKF/go-utility/v2/uuid"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"gopkg.in/yaml.v3"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/SKF/go-enlight-middleware/client-id/models"
+)
+
+type clientMock struct {
+	mock.Mock
+}
+
+func (mock *clientMock) GetObjectWithContext(ctx context.Context, inp *s3.GetObjectInput, opt ...request.Option) (*s3.GetObjectOutput, error) {
+	args := mock.Called(ctx, inp, opt)
+	return args.Get(0).(*s3.GetObjectOutput), args.Error(1)
+}
+
+func TestS3GetClientID_Happy(t *testing.T) {
```

Comment:
This is quite a long test method. maybe it could be easier to see what is actually tested here by some function extraction or maybe using the pattern with //ARRANGE //ACT //ASSERT comments

Reply:
Was thinking of writing some helpers, but didn't since I only wrote one test. Once you start writing more tests it will be easier to see what setup will be duplicated.